model/task_lifecycle.go

```diff
@@ -842,11 +845,20 @@ func UpdateBuildAndVersionStatusForTask(taskId string, updates *StatusChanges) e
 		return err
 	}
 
-	if b.Status == evergreen.BuildCreated {
+	v, err := VersionFindOneId(b.Version)
+	if err != nil {
+		return errors.Wrap(err, "error finding version")
+	}
+	if v == nil {
+		return errors.Errorf("version '%s' not found", b.Version)
+	}
+
+	if b.Status == evergreen.BuildCreated || unfinishedTasks > 0 {
```

Comment:
Should we also check that finishedTasks != 0 so we don't update twice, now that we're considering a new case?

Reply:
I don't think that's quite correct, because this should be hit when there are no finished tasks but a task has started. I think in practice this will be fine because it's only called when a task ends or is reset

Reply:
Then I guess should line 915 consider if there are unfinishedTasks? The case you're describing will update Version/Build as started, and then hit 915 and update as Created.

Reply:
That doesn't work either because unfinishedTasks does not differentiate between started and unstarted. The case where it matters doesn't need to be handled this doesn't get called when tasks start. I could add handling for it, but that would make things messier

Reply:
Sorry, can you explain this further? I don't understand this sentence: "The case where it matters doesn't need to be handled this doesn't get called when tasks start."

Are you saying that the case where this is updated twice doesn't exist? Or that it's fine to update twice?

Reply:
Mostly the first. This function is only called when a task finishes or restarts, so you would need to individually restart all tasks in a build for this to happen. That scenario is more realistic than the scenario where adding the check causes a problem (calling this function while a task is running but no tasks have finished) so I made the change